src/main/java/com/adobe/aem/compgenerator/web/ConfigurationReadWriteServlet.java

```diff
@@ -0,0 +1,359 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static com.adobe.aem.compgenerator.Constants.*;
+
+/**
+ * ConfigurationReadWriteServlet
+ *
+ * contains the methods to update the various
+ * fields within the GenerationConfig class
+ * and to generate the code on demand from the web UI
+ *
+ */
+public class ConfigurationReadWriteServlet extends HttpServlet {
+    private static final Logger LOG = LogManager.getLogger(ConfigurationReadWriteServlet.class);
+
+    @Override
+    public void init(final ServletConfig config) throws ServletException {
+        super.init(config);
+    }
+
+    @Override
+    /*
+     * Handles retrieving the current state of the configuration json
+     */
+    protected void doGet(final HttpServletRequest req, final HttpServletResponse resp) throws ServletException, IOException {
+        resp.setContentType("application/json; charset=UTF-8");
+        resp.setCharacterEncoding("UTF-8");
+        File configFile = new File(CONFIG_PATH);
+        GenerationConfig config = CommonUtils.getComponentData(configFile);
+
+        ObjectMapper mapper = new ObjectMapper();
+
+        PrintWriter writer = resp.getWriter();
+        writer.write(mapper.writeValueAsString(config));
+        writer.close();
+    }
+
+    @Override
+    /*
+     * Handles resetting the config file to a default state
+     */
+    protected void doDelete(final HttpServletRequest req, final HttpServletResponse resp) throws ServletException, IOException {
+        resp.setContentType("application/json; charset=UTF-8");
+        resp.setCharacterEncoding("UTF-8");
+        File configFile = new File(CONFIG_PATH);
+        ObjectMapper mapper = new ObjectMapper();
+        Message msg = new Message(false, "");
+        if (configFile.exists()) {
+            boolean d = configFile.delete();
+            if (d) {
+                // reset the config file to defaults
+                String configEmptyPath = "data-config-empty.json";
+                try {
+                    InputStream input = AemCompGenerator.class.getResourceAsStream("/resources/" + configEmptyPath);
+                    if (input == null) {
+                        input = AemCompGenerator.class.getClassLoader().getResourceAsStream(configEmptyPath);
```

Comment:
Is there a reason we need to look in two separate places for this, given that it's part of the compiled artifact and thus should be dependable?

Reply:
This alternate path is due to the difference in where the static files are located when running the project from a JAR vs running the project within an IDE. It could be done a bit differently or could use a comment explaining as such.

Reply:
ah ya, I think some comments to clarify what this code is accomplishing would definitely help.